app/src/main/java/com/android/shelfLife/viewmodel/overview/HouseholdCreationScreenViewModel.kt

```diff
@@ -3,131 +3,166 @@ package com.android.shelfLife.viewmodel.overview
 import android.util.Log
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.android.shelfLife.model.newFoodItem.FoodItemRepository
 import com.android.shelfLife.model.newInvitations.InvitationRepository
 import com.android.shelfLife.model.newhousehold.HouseHold
 import com.android.shelfLife.model.newhousehold.HouseHoldRepository
 import com.android.shelfLife.model.user.UserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 
-class HouseholdCreationScreenViewModel(
+@HiltViewModel
+class HouseholdCreationScreenViewModel
+@Inject
+constructor(
     private val houseHoldRepository: HouseHoldRepository,
+    private val foodItemRepository: FoodItemRepository,
     private val invitationRepository: InvitationRepository,
     private val userRepository: UserRepository
 ) : ViewModel() {
+
   private val _emailList = MutableStateFlow<Set<String>>(emptySet())
   val emailList: StateFlow<Set<String>> = _emailList.asStateFlow()
 
-  val householdToEdit = houseHoldRepository.householdToEdit
-  val selectedHousehold = userRepository.selectedHousehold
-  val households = houseHoldRepository.households
-  val currentUser = userRepository.user
+  val householdToEdit: StateFlow<HouseHold?> =
+      houseHoldRepository.householdToEdit.stateIn(
+          viewModelScope, started = SharingStarted.Eagerly, null)
+
+  val selectedHousehold =
+      houseHoldRepository.selectedHousehold.stateIn(
+          viewModelScope, started = SharingStarted.Eagerly, null)
+
```

Comment:
Great job initialising these variables with a sharing started eagerly => means these will get populated and we won't have unnecessary lag